lib/Doctrine/DBAL/Types/ConversionException.php

```diff
@@ -65,4 +65,29 @@ static public function conversionFailedFormat($value, $toType, $expectedFormat)
             $toType . '. Expected format: ' . $expectedFormat
         );
     }
+
+    /**
+     * Thrown when the PHP value passed to the converter was not of the expected type.
+     *
+     * @param mixed $value
+     * @param string $fromType
+     *
+     * @return \Doctrine\DBAL\Types\ConversionException
+     */
+    static public function conversionFailedInvalidType($value, $toType, $fromType)
+    {
+        $actualType = gettype($value);
+        if ($actualType === 'object') {
+            $actualType .= " (" . get_class($value) . ")";
+            if (!method_exists($value, '__toString')) {
+                $value = 'object';
+            }
+        }
+        $value = (string)$value;
```

Comment:
Argh, I just realised that this logic is too convoluted :-(

@zeroedin-bill:

 - arrays will crash this
 - testing is needed
 - I simply wouldn't care about abbreviating the string
 - I wouldn't rely on `__toString`